### DISCUSSION TITLE
Recurring tasks are disapeared (or are deleted) from DB table "scheduled_tasks" after executing.

### DISCUSSION THREAD

**Discussion by ellalalaliu** [kagkarlsson/db-scheduler#408] [Q&A]
Hi Kagkarlsson and community,

I use the db-scheduler in a web application for executing 38 Recurring tasks.
All of tasks are "Static recurring task" and they are arranged by different schedule.
For Example:
  @Bean
  RecurringTask<Void> weekDownloadAlarm(WeekDownloadAlarmJob myService) {
    RecurringTask<Void> job = null;

    String taskName = "admin-weekly-pd-download-alarm";
    String cronPattern = "0 30 7 ? * MON";

    job = Tasks.recurring(taskName, new CronSchedule(cronPattern)).execute((instance, ctx) -> {
      log.info(taskName + " Instance: {}, ctx: {}", instance, ctx);
      myService.weekDownloadAlarm();
    });

    return job;
  }

All of the configuration options keep as default:
db-scheduler.enabled=true
db-scheduler.heartbeat-interval=5m
db-scheduler.polling-interval=10s
db-scheduler.polling-limit=
db-scheduler.table-name=scheduled_tasks
db-scheduler.immediate-execution-enabled=false
db-scheduler.scheduler-name=
db-scheduler.threads=10
db-scheduler.delay-startup-until-context-ready=false
db-scheduler.polling-strategy=fetch
db-scheduler.polling-strategy-lower-limit-fraction-of-threads=0.5
db-scheduler.polling-strategy-upper-limit-fraction-of-threads=3.0
db-scheduler.shutdown-max-wait=30m

The 38 Recurring tasks work well in the testing system environment.
However, there is a weird situation happening in the production environment.
Sometimes, some tasks are disappeared (or are deleted) from DB table "scheduled_tasks" after executing.
It is not fixed time and not fixed tasks so I have to check the DB table "scheduled_tasks" from time to time, in order to restart the web application.

The only difference between the testing and production system environment is the amount of data in the database.
I might need to adjust the configuration options for the production system environment.  Could you give me advice or any other suggestion?

Thank you.

**Comment by kagkarlsson** [comment]
Is there any chance you have scheduler-instances using the same table but with different task-sets? This might happen then due to the `deleteUnresolved` feature.

```
⚙️ .deleteUnresolvedAfter(Duration)
The time after which executions with unknown tasks are automatically deleted. These can typically be old recurring tasks that are not in use anymore. This is non-zero to prevent accidental removal of tasks through a configuration error (missing known-tasks) and problems during rolling upgrades. Default 14d.
```

You might find traces of the following in your logs:

```
      LOG.info(
          "Found execution with unknown task-name '{}'. Adding it to the list of known unresolved task-names.",
          taskName);
```

or 

```
LOG.warn(
                  "Deleting all executions for task with name '{}'. They have been unresolved for more than {}",
                  taskName,
                  deleteUnresolvedAfter);
```
